### PR TITLE
[pytorch] simplify android circleci definition data model

### PR DESCRIPTION
**Pull request by ljk53**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40010 [pytorch] upload android build size to scuba
* **#40007 [pytorch] simplify android circleci definition data model**

Summary:
Android jobs don't seem to fit to `pytorch_build_data.py` data model
very well. Other mobile jobs all have their own data model files - even
for Android nightly jobs. As we are adding more variants like vulkan,
it's going to be hard to maintain.

So renamed `android_gradle.py` to `android_definitions.py` and moved
android jobs into it, following the conventions of `nightly_android.py`
and `ios_definitions.py`.

Differential Revision: [D22036915](https://our.internmc.facebook.com/intern/diff/D22036915)

### CODE REVIEW

File: .circleci/cimodel/data/simple/android_gradle.py [hunk 1]
```diff
@@ -1,37 +0,0 @@
...
-            "name": self.job_name,
-            "requires": self.dependencies,
-        }
-
-        return [{self.template_name: props_dict}]
-
-
-WORKFLOW_DATA = [
-    AndroidJob(
-        "pytorch-linux-xenial-py3-clang5-android-ndk-r19c-gradle-build-x86_32",
```
Comment (ljk53) [inline .circleci/cimodel/data/simple/android_gradle.py:23]: The current code sets master branch filter on android gradle jobs unconditionally - I think this job should not have master branch filter. It's supposed to run on PR but not run on master commits: https://github.com/pytorch/pytorch/blob/master/.circleci/verbatim-sources/job-specs/job-specs-custom.yml#L284
With the current config the job doesn't run at all. In this PR I added `is_master_only=False` for this job. If we disabled the job on PR for good reason (too flaky?) then we should consider deleting it.

File: .circleci/cimodel/data/simple/android_definitions.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import cimodel.data.simple.util.branch_filters
+from cimodel.data.simple.util.docker_constants import DOCKER_IMAGE_NDK
+
+
+class AndroidJob:
```
Comment (ljk53) [inline .circleci/cimodel/data/simple/android_definitions.py:5]: This is similar to https://github.com/pytorch/pytorch/blob/master/.circleci/cimodel/data/simple/nightly_android.py
Maybe we should consider merging these definitions in the future.

**Review by IvanKobzarev — APPROVED** [review]

**Comment by ezyang** [comment]
I verified that every config in the old yaml was present in the new yaml. I will let @seemethere review for code structure in generation scripts.

**Comment by ljk53** [comment]
> I verified that every config in the old yaml was present in the new yaml. I will let @seemethere review for code structure in generation scripts.

oops, I already landed :)
I believe the functionality is the same as before (except I removed a master filter that I don't think should be added)
@seemethere / @kostmo please let me know if you have concern with this change.

**Pull request closed without merging**